superset-frontend/src/SqlLab/actions/sqlLab.js

```diff
@@ -869,14 +870,32 @@ export function saveQuery(query) {
           query,
           result: convertQueryToClient(result.json.item),
         });
-        dispatch(addSuccessToast(t('Your query was saved')));
         dispatch(queryEditorSetTitle(query, query.title));
+        return convertQueryToClient(result.json.item);
```

Comment:
> @betodealmeida we need to return it so that we can used the savedQuery context here, to pass it into the next action.

You should compute it once then, and use it here and in line 871, instead of computing it twice.